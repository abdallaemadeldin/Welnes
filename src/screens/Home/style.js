import { StyleSheet } from 'react-native';
import { useSafeAreaInsets } from 'react-native-safe-area-context';

export const style = () => {
    const { bottom } = useSafeAreaInsets();

    return StyleSheet.create({
        container: {
            flex: 1,
            backgroundColor: '#ccc',
            alignItems: 'center'
        },
        headerView: {
            width: '100%',
            flexDirection: 'row',
            justifyContent: 'center',
            alignItems: 'center',
            paddingVertical: 20
        },
        btn: {
            flex: 1,
            marginHorizontal: 10,
            backgroundColor: '#fff',
            justifyContent: 'center',
            alignItems: 'center',
            height: 44,
            borderRadius: 4
        },
        videoCard: {
            flex: 1,
            height: 200,
            marginHorizontal: 10,
            backgroundColor: '#fff',
            borderRadius: 4,
            marginBottom: 10
        },
        list: {
            width: '100%'
        },
        cardOverlay: {
            flex: 1,
            justifyContent: 'center',
            alignItems: 'center',
            backgroundColor: '#0008'
        },
        playIcon: {
            width: 70,
            height: 70,
            tintColor: '#fff',
        },
        empty: {
            textAlign: 'center',
            marginTop: '10%',
            fontWeight: 'bold',
            fontSize: 18,
            color: '#000'
        },
        uploadingOverlay: {
            position: 'absolute',
            top: 0,
            left: 0,
            bottom: 0,
            right: 0,
            backgroundColor: '#0005',
            justifyContent: 'center',
            alignItems: 'center',
            zIndex: 1
        },
        uploadingText: {
            fontSize: 16,
            color: '#fff',
            lineHeight: 26,
            fontWeight: 'bold'
        },
        contentContainerStyle: {
            paddingBottom: bottom + 10
        },
        indicator: {
            marginTop: '30%'
        }
    });
}