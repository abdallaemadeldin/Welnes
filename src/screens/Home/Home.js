import React, { memo } from 'react';
import { FlatList, Text, TouchableOpacity, View, Image, ImageBackground } from 'react-native';
import { style } from './style';

const Home = () => {
    const { container, headerView, btn, videoCard, list, playIcon, cardOverlay } = style();

    const renderItem = () => {
        return (
            <ImageBackground source={{ uri: 'https://www.socialmediaexaminer.com/wp-content/uploads/2015/06/ds-video-thumbnails-480.png' }} style={videoCard}>
                <TouchableOpacity activeOpacity={1} style={cardOverlay}>
                    <Image source={require('src/assets/play.png')} style={playIcon} />
                </TouchableOpacity>
            </ImageBackground>
        );
    }

    return (
        <View style={container}>
            <View style={headerView}>
                <TouchableOpacity style={btn} activeOpacity={.8}>
                    <Text>Select</Text>
                </TouchableOpacity>
                <TouchableOpacity style={btn} activeOpacity={.8}>
                    <Text>Capture</Text>
                </TouchableOpacity>
            </View>

            <FlatList
                data={[{}, {}, {}]}
                renderItem={renderItem}
                keyExtractor={(_, i) => i.toString()}
                showsVerticalScrollIndicator={false}
                bounces={false}
                style={list}
            />
        </View>
    );
}

export default memo(Home);