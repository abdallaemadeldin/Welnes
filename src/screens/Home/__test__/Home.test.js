import React from 'react';
import renderer from 'react-test-renderer';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import Home from '../Home';

jest.mock("react-native-fs", () => "RNFS");
jest.mock("react-native-image-crop-picker", () => "RNImagePicker");
jest.mock("react-native-ffmpeg", () => "FFMPEG");

jest.mock("@react-navigation/native", () => {
    const actualNav = jest.requireActual("@react-navigation/native");
    return {
        ...actualNav,
        useNavigation: () => ({
            replace: jest.fn(),
        }),
    };
});

it('renders correctly', () => {
    const tree = renderer.create(
        <SafeAreaProvider initialMetrics={{ insets: { top: 0, bottom: 0, left: 0, right: 0 } }}>
            <Home />
        </SafeAreaProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});