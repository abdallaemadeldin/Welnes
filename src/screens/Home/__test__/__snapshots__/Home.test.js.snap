// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ccc",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingVertical": 20,
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "flex": 1,
            "height": 44,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "opacity": 1,
          }
        }
      >
        <Text>
          Select
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "flex": 1,
            "height": 44,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "opacity": 1,
          }
        }
      >
        <Text>
          Capture
        </Text>
      </View>
    </View>
    <ActivityIndicator
      color="#000"
      style={
        Object {
          "marginTop": "30%",
        }
      }
    />
  </View>
</RNCSafeAreaProvider>
`;
